baselines/niid_bench/niid_bench/main.py

```diff
@@ -0,0 +1,123 @@
+        cfg.model.input_dim = 256
+        cfg.model._target_ = 'niid_bench.models.CNNMnist'
+    print(OmegaConf.to_yaml(cfg))
+
+    # 2. Prepare your dataset
+    # here you should call a function in datasets.py that returns whatever is needed to:
+    # (1) ensure the server can access the dataset used to evaluate your model after
+    # aggregation
+    # (2) tell each client what dataset partitions they should use (e.g. a this could
+    # be a location in the file system, a list of dataloader, a list of ids to extract
+    # from a dataset, it's up to you)
+    trainloaders, valloaders, testloader = load_datasets(
+        config=cfg.dataset,
+        num_clients=cfg.num_clients,
+        val_ratio=cfg.dataset.val_split,
+    )
+
+    # 3. Define your clients
+    # Define a function that returns another function that will be used during
+    # simulation to instantiate each individual client
+    client_fn = None
+    if cfg.client_fn._target_ == "niid_bench.client_scaffold.gen_client_fn":
+        save_path = HydraConfig.get().runtime.output_dir
+        client_cv_dir = os.path.join(save_path, "client_cvs")
+        print("Local cvs for scaffold clients are saved to: ",  client_cv_dir)
+        client_fn = call(cfg.client_fn,
+            trainloaders,
+            valloaders,
+            model=cfg.model,
+            client_cv_dir=client_cv_dir,
+        )
+    else:
+        client_fn = call(cfg.client_fn, 
+            trainloaders,
+            valloaders,
+            model=cfg.model,
+        )
+
+    device = cfg.server_device
+    evaluate_fn = gen_evaluate_fn(testloader, device=device, model=cfg.model)
+
+    # 4. Define your strategy
+    # pass all relevant argument (including the global dataset used after aggregation,
+    # if needed by your method.)
+    # strategy = instantiate(cfg.strategy, <additional arguments if desired>)
+    strategy = instantiate(
+        cfg.strategy,
+        evaluate_fn=evaluate_fn,
+    )
+
+    # 5. Define your server
+    server = Server(strategy=strategy, client_manager=SimpleClientManager())
+    if isinstance(strategy,FedNovaStrategy):
+        server = FedNovaServer(strategy=strategy, client_manager=SimpleClientManager())
+    elif isinstance(strategy,ScaffoldStrategy):
+        server = ScaffoldServer(strategy=strategy, model=cfg.model, client_manager=SimpleClientManager())
+
+    # 6. Start Simulation
+    # history = fl.simulation.start_simulation(<arguments for simulation>)
+    history = fl.simulation.start_simulation(
+        server=server,
+        client_fn=client_fn,
+        num_clients=cfg.num_clients,
+        config=fl.server.ServerConfig(num_rounds=cfg.num_rounds),
+        client_resources={
+            "num_cpus": cfg.client_resources.num_cpus,
+            "num_gpus": cfg.client_resources.num_gpus,
+        },
+        strategy=strategy,
+    )
+
+    print(history)
+
+    save_path = HydraConfig.get().runtime.output_dir
+    print(save_path)
```

Comment:
Could you add just a couple of lines of code to write a pickle to `save_path` (i.e. the output directory for the experiment) with there content in the `history` object ? Other baselines do this, it is useful to have if people want to create plots later on easily.